src/go/intelligence/int_test/test_wrapper.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+##H Script for automation of testing process of the intelligence module.
+
+#Variables
+CMSMONITORING_REPO=$(pwd)/CMSMonitoring
+AM_BIN=$(pwd)/am/alertmanager
+TEST_CONFIG=$(pwd)/CMSMonitoring/src/go/intelligence/int_test/test_config.json
+AM_VERSION="alertmanager-0.21.0.linux-amd64"
+AM_URL="https://github.com/prometheus/alertmanager/releases/download/v0.21.0/$AM_VERSION.tar.gz"
+
+PID=$(ps auxwww | egrep "alertmanager" | grep -v grep | awk 'BEGIN{ORS=" "} {print $2}')
+
+# function for starting AlertManager
+start_am() {
+    echo "Starting AlertManager in background."
+    nohup ./am/alertmanager --config.file=./am/alertmanager.yml </dev/null 2>&1 >AM.log &
+}
+
+# function for stopping AlertManager
+stop_am() {
+    local PID=$(ps auxwww | egrep "alertmanager" | grep -v grep | awk 'BEGIN{ORS=" "} {print $2}')
+    echo "Stopping Alertmanager. PID : ${PID}"
+    if [ -n "${PID}" ]; then
+        kill -9 ${PID}
+    fi
+}
+
+# Logic for deploying the alertmanager
+if [ ! -z "$PID" ]; then
+    echo "AlertManager already running. PID : ${PID}"
+else
+    if [ -f "$AM_BIN" ]; then
+        start_am
+    else
+        echo "AlertManager not found !!"
+        if [ -x "$(command -v wget)" ]; then
+            echo "Downloading AlertManager...."
+            wget $AM_URL -O am.tar.gz
```

Comment:
here you assume that wget is in user path, it may not be the case. Please check that it exists first.

Reply:
Yes Valentin that's what  [ -x "$(command -v wget)" ] this is checking, if wget is there it will download otherwise it will annotate the user about it. You can check in the else section. 
echo "Install tar to continue. Exiting.."

Reply:
ok, I missed the command part. But it will leave a test workflow half way through, i.e. if wget failed the previous steps will be executed and orphans. Please improve error handling to instruct users (a) where to look for unfinished installation, (b) how to fix the problem, ( c) how to clean-up unfinished installation., etc.

Reply:
Yes, Valentin. I have checked for it. Now the first Directory is passed and checked:-
- If available - OK
- If not available - CREATE
- if no permission - EXIT with a message

Now once the working directory is created it will try to check
- AM running - MOVE ahead
- AM not running - CHECK available in $WDIR and run
- if AM not available - Check Wget 
- if Wget available - download the AM and run it.
- if Wget not available - deletes $WDIR and exits with a message

now int module is build
- if go not available - delete $WDIR and exits with a message
- if go available - build and run test , prints the result, stops AM, deletes $WDIR